packages/apps/graph-client/src/components/extended-transactions-table/extended-transactions-table.tsx

```diff
@@ -0,0 +1,130 @@
+import { Box, Button, Link, Table } from '@kadena/react-ui';
+
+import type { GetTransactionsQuery } from '@/__generated__/sdk';
+import routes from '@/constants/routes';
+import type { FetchMoreOptions, FetchMoreQueryOptions } from '@apollo/client';
+import React from 'react';
+
+type DataType = GetTransactionsQuery;
+
+interface IVariableType {
+  first: number | null;
+  last: number | null;
+  after: string | null;
+  before: string | null;
+}
+
+interface IExpandedTransactionsTableProps {
+  transactions: GetTransactionsQuery['transactions'];
+  fetchMore: (
+    fetchMoreOptions: FetchMoreQueryOptions<IVariableType, DataType> &
+      FetchMoreOptions,
+  ) => Promise<any>;
+}
+
+export const ExtendedTransactionsTable = (
+  props: IExpandedTransactionsTableProps,
+): JSX.Element => {
+  const { transactions, fetchMore } = props;
+
+  return (
+    <>
+      <Box marginBottom="$3">
+        <span>Showing 10 results per page</span>
+
+        <Button
+          variant="compact"
+          onClick={() =>
+            fetchMore({
+              variables: {
+                first: 20,
+                last: null,
+                after: transactions.pageInfo.endCursor,
+                before: null,
+              },
+              updateQuery: (prev, { fetchMoreResult }) => {
+                if (!fetchMoreResult) return prev;
+                return fetchMoreResult;
+              },
+            })
+          }
+          disabled={!transactions.pageInfo.hasNextPage}
+          style={{ float: 'right' }}
+        >
+          Next Page
+        </Button>
+        <Button
+          variant="compact"
+          onClick={() =>
+            fetchMore({
+              variables: {
+                first: null,
+                last: 20,
+                after: null,
+                before: transactions.pageInfo.startCursor,
+              },
+              updateQuery: (prev, { fetchMoreResult }) => {
+                if (!fetchMoreResult) return prev;
+
+                if (fetchMoreResult.transactions.edges.length < 10) {
+                  return {
+                    ...prev,
+                    transactions: {
+                      ...fetchMoreResult.transactions,
+                      edges: [
+                        ...fetchMoreResult.transactions.edges,
+                        ...prev.transactions.edges,
+                      ].slice(0, 10),
```

Comment:
Shouldn't this '10' and the 10 above be 20 instead, since you get the first and last 20 int he fetch more?

Reply:
Yep, it should. I was already proceeding with the changes for selectability on table size and pagination but thought I'd leave that for another PR. I'll leave it 10 for this one